proto/build.rs

```diff
@@ -11,131 +11,151 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::env;
-use std::fs::{self, File};
-use std::io::Write;
-use std::path::{Path, PathBuf};
-
-use grpcio_compiler::codegen as grpc_gen;
-use prost::Message;
-use prost_types::FileDescriptorSet as ProstFileDescriptorSet;
-use protobuf::compiler_plugin::GenResult;
-use protobuf::descriptor::{FileDescriptorProto, FileDescriptorSet};
-use protobuf_codegen::{self as pb_gen, Customize};
-
-fn write_files<W: Write>(
-    results: impl Iterator<Item = (String, Vec<u8>)>,
-    output_dir: &Path,
-    mut out: W,
-) {
-    if !output_dir.exists() {
-        fs::create_dir_all(output_dir).unwrap();
+fn main() {
+    for package in &["testing", "example", "health"] {
+        protobuf::compile(format!("{}.desc", package), package);
+        prost::compile(format!("{}.desc", package));
     }
+}
 
-    for (name, content) in results {
-        let out_file = output_dir.join(&name);
-        let mut f = File::create(&out_file).unwrap();
-        f.write_all(&content).unwrap();
-        let (module_name, _) = name.split_at(name.len() - 3); // ".rs".len() == 3
-        writeln!(out, "pub mod {};", module_name).unwrap();
+#[cfg(feature = "prost-codec")]
```

Comment:
Why do you opt for this over `cfg(feature ...)`'ing in the `main()`?

Reply:
because that alternative means the functions must compile even if they are not used, which means that if using the protobuf feature, we would still need to depend on Prost